FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/report_builder.c

```diff
@@ -323,6 +362,84 @@ static eReportBuilderStatus prvWriteConnectionsArray( char * pcBuffer,
 }
 /*-----------------------------------------------------------*/
 
+static eReportBuilderStatus prvWriteTaskIdsArray( char * pcBuffer,
+                                                  uint32_t ulBufferLength,
+                                                  const uint32_t * pulTaskIdsArray,
+                                                  uint32_t pulTaskIdsArrayLength,
+                                                  uint32_t * pulOutCharsWritten )
+{
+    char * pcCurrentWritePos = pcBuffer;
+    uint32_t i, ulRemainingBufferLength = ulBufferLength;
+    int32_t lCharactersWritten;
+    eReportBuilderStatus eStatus = eReportBuilderSuccess;
+
+    configASSERT( pcBuffer != NULL );
+    configASSERT( pulTaskIdsArray != NULL );
+    configASSERT( pulOutCharsWritten != NULL );
+
+    /* Write the JSON array open marker. */
+    if( ulRemainingBufferLength > 1 )
+    {
+        *pcCurrentWritePos = reportbuilderJSON_ARRAY_OPEN_MARKER;
+        ulRemainingBufferLength -= 1;
+        pcCurrentWritePos += 1;
+    }
+    else
+    {
+        eStatus = eReportBuilderBufferTooSmall;
+    }
+
+    /* Write the array elements. */
+    for( i = 0; ( ( i < pulTaskIdsArrayLength ) && ( eStatus == eReportBuilderSuccess ) ); i++ )
+    {
+        lCharactersWritten = snprintf( pcCurrentWritePos,
+                                       ulRemainingBufferLength,
+                                       "%u,",
+                                       pulTaskIdsArray[ i ] );
+
+        if( !reportbuilderSNPRINTF_SUCCESS( lCharactersWritten, ulRemainingBufferLength ) )
+        {
+            eStatus = eReportBuilderBufferTooSmall;
+            break;
```

Comment:
This line isn't needed since you're testing `( eStatus == eReportBuilderSuccess )` on line 393 – delete it.